challenge.rb

```diff
@@ -1,26 +1,32 @@
 
 def capitalize_each_string(input)
   #implement your solution here
+  input.map {|s| s.capitalize}
 end
 
 def fetch_the_dog(input)
   #implement your solution here
+  input.find_all {|s| s.downcase == "dog"}
 end
 
 def no_dogs_allowed(input)
   #implement your solution here
+  input.reject {|s| s.downcase == "dog"}
 end
 
 def count_the_animals(input)
   #implement your solution here
+  input.count
 end
 
 def fetch_the_first_two(input)
   #implement your solution here
+  input.take(2)
 end
 
 def fetch_CD_animals(input)
   #implement your solution here
+  input.select{|s| s.downcase.chars[0] == 'c' || s.downcase.chars[0] == 'd'}
```

Comment:
You can simplify this with a regular expression.  Google `rubular` for a nice tool for this.